src/arrayMethodReduce.js

```diff
@@ -4,8 +4,37 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, initialValue = '') {
+    let sum = initialValue;
+    // if given array contains only strings
+    let startValue = '';
+
+    if (typeof this[0] === 'number') {
+      startValue = 0;
+    }
+
+    // when passed two arguments and equal to undefined
+    if (arguments.length === 2 && arguments[1] === undefined) {
```

Comment:
the second argument will never be undefined unless you pass it explicitly, which you never wanna do. you have set the default value for `initialValue`, that's why it will never be undefined